ddl/placement/bundle.go

```diff
@@ -49,6 +50,200 @@ func NewBundle(id int64) *Bundle {
 	}
 }
 
+// NewBundleFromConstraintsOptions will transform constraints options into the bundle.
+func NewBundleFromConstraintsOptions(options *model.PlacementSettings) (*Bundle, error) {
+	if options == nil {
+		return nil, fmt.Errorf("%w: options can not be nil", ErrInvalidPlacementOptions)
+	}
+
+	if len(options.PrimaryRegion) > 0 || len(options.Regions) > 0 || len(options.Schedule) > 0 {
+		return nil, fmt.Errorf("%w: should be [LEADER/VOTER/LEARNER/FOLLOWER]_CONSTRAINTS=.. [VOTERS/FOLLOWERS/LEARNERS]=.., mixed other sugar options %s", ErrInvalidPlacementOptions, options)
+	}
+
+	constraints := options.Constraints
+	leaderConstraints := options.LeaderConstraints
+	learnerConstraints := options.LearnerConstraints
+	followerConstraints := options.FollowerConstraints
+	voterConstraints := options.VoterConstraints
+	followerCount := options.Followers
+	voterCount := options.Voters
+	learnerCount := options.Learners
+
+	if voterCount+followerCount == 0 {
+		return nil, fmt.Errorf("%w: not enough voter, please specify voter or follower count", ErrInvalidPlacementOptions)
+	}
+
+	CommonConstraints, err := NewConstraintsFromYaml([]byte(constraints))
+	if err != nil {
+		return nil, fmt.Errorf("%w: 'Constraints' should be [constraint1, ...] or any yaml compatible array representation", err)
+	}
+
+	Rules := []*Rule{}
+
+	LeaderConstraints, err := NewConstraintsFromYaml([]byte(leaderConstraints))
```

Comment:
similar to `NewMergeRules` , can we reuse that? anyway, i'm not strong on this

Reply:
Actually, I limited `leader_constraints` to be a list, too. `NewConstraintsFromYaml` is for array/list only. We don't need any extra check.

As for reusing, I don't think it is worthy. We could do the check along with the construction process. Checking again before the construction seems redundant.

Reply:
One more thing, `NewMergeRules` is for at least two constraints. Here we only parse one constraint. And that function invokes `yaml.Unmarshal` again and again, not so efficient.

Reply:
If we really want to reuse, replace `NewMergeRules` with `NewBundleFromOptions` seems more reasonable and efficient.

Reply:
Small bench:

```
BenchmarkMerger-16        112108             14678 ns/op
BenchmarkNew-16          1527146               765.6 ns/op
```